src/mbgl/renderer/image_manager.hpp

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include <mbgl/style/image_impl.hpp>
+#include <mbgl/renderer/image_atlas.hpp>
+#include <mbgl/util/noncopyable.hpp>
+#include <mbgl/util/immutable.hpp>
+#include <mbgl/util/optional.hpp>
+#include <mbgl/gl/texture.hpp>
+
+#include <mapbox/shelf-pack.hpp>
+
+#include <set>
+#include <string>
+
+namespace mbgl {
+
+namespace gl {
+class Context;
+} // namespace gl
+
+class ImageRequestor {
+public:
+    virtual ~ImageRequestor() = default;
+    virtual void onImagesAvailable(ImageMap) = 0;
+};
+
+/*
+    ImageManager does two things:
+
+        1. Tracks requests for icon images from tile workers and sends responses when the requests are fulfilled.
+        2. Builds a texture atlas for pattern images.
+
+    These are disparate responsibilities and should eventually be handled by different classes. When we implement
+    data-driven support for `*-pattern`, we'll likely use per-bucket pattern atlases, and that would be a good time
+    to refactor this.
```

Comment:
The `*-pattern` properties are all paint properties and not currently DDS enabled. So there's only one possible pattern per layer, and changing it via runtime styling takes effect immediately, without an async layout step. If/when we enable DDS, then we probably will want per-bucket atlases (at least in the data-driven case), but for now it makes more sense to keep it global.